hal/riscv64/string.c

```diff
@@ -118,7 +88,7 @@ static inline char *hal_strcpy(char *dest, const char *src)
 }
 
 
-static inline char *hal_strncpy(char *dest, const char *src, size_t n)
+char *hal_strncpy(char *dest, const char *src, size_t n)
 {
 	int i = 0;
```

Comment:
What if `n==0`?

Reply:
Yeah,  good remark. This function has been updated.